test/Microsoft.AspNet.Routing.Tests/RouteCollectionTest.cs

```diff
@@ -231,6 +232,300 @@ public void NamedRouteTests_GetNamedRoute_AmbiguousRoutesInCollection_ThrowsForA
             Assert.Equal("The supplied route name 'ambiguousRoute' is ambiguous and matched more than one route.", ex.Message);
         }
 
+        [Fact]
+        public void GetVirtualPath_AmbiguousRoutes_RequiresRouteValueValidation_Error()
+        {
+            // Arrange
+            var namedRoute = CreateNamedRoute("Ambiguous", accept: false);
+
+            var routeCollection = new RouteCollection();
+            routeCollection.Add(namedRoute);
+
+            var innerRouteCollection = new RouteCollection();
+            innerRouteCollection.Add(namedRoute);
+            routeCollection.Add(innerRouteCollection);
+
+            var options = new RouteOptions()
+            {
+                UseBestEffortLinkGeneration = true,
+            };
+
+            var virtualPathContext = CreateVirtualPathContext("Ambiguous", options: options);
+
+            // Act & Assert
+            var ex = Assert.Throws<InvalidOperationException>(() => routeCollection.GetVirtualPath(virtualPathContext));
+            Assert.Equal("The supplied route name 'Ambiguous' is ambiguous and matched more than one route.", ex.Message);
+        }
+
+        [Fact]
+        public void GetVirtualPath_NamedRoute_BestEffort_BestInTopCollection()
+        {
+            // Arrange
+            var bestMatch = CreateNamedRoute("Match", accept: true, matchValue: "best");
+            var noMatch = CreateNamedRoute("NoMatch", accept: true, matchValue: "bad");
+
+            var routeCollection = new RouteCollection();
+            routeCollection.Add(noMatch);
```

Comment:
is this intended? I thought we would add best route in top collection. there seems to be no difference between GetVirtualPath_NamedRoute_BestEffort_BestInTopCollection and GetVirtualPath_NamedRoute_BestEffort_BestMatchInNestedCollection